embassy-usb-logger/src/lib.rs

```diff
@@ -105,6 +108,38 @@ impl<const N: usize> UsbLogger<N> {
             join(run_fut, join(log_fut, discard_fut)).await;
         }
     }
+
+    /// Creates the futures needed for the logger from a given class
+    /// This can be used in cases where the usb device is already in use for another connection
+    pub async fn create_future_from_class<'d, D>(&'d self, class: CdcAcmClass<'d, D>)
+    where
+        D: Driver<'d>,
+    {
+        const MAX_PACKET_SIZE: u8 = 64;
+        let (mut sender, mut receiver) = class.split();
+
+        loop {
+            let log_fut = async {
+                let mut rx: [u8; MAX_PACKET_SIZE as usize] = [0; MAX_PACKET_SIZE as usize];
+                sender.wait_connection().await;
+                loop {
+                    let len = self.buffer.read(&mut rx[..]).await;
+                    let _ = sender.write_packet(&rx[..len]).await;
+                    if len as u8 == MAX_PACKET_SIZE {
+                        let _ = sender.write_packet(&[]).await;
+                    }
+                }
+            };
+            let discard_fut = async {
+                let mut discard_buf: [u8; MAX_PACKET_SIZE as usize] = [0; MAX_PACKET_SIZE as usize];
+                receiver.wait_connection().await;
+                loop {
+                    let _ = receiver.read_packet(&mut discard_buf).await;
+                }
+            };
+            join(log_fut, discard_fut).await;
+        }
```

Comment:
What if you make this an internal fn that takes a &mut sender and &mut receiver in order to reuse code?

Reply:
Great idea! Done